pkg/sql/create.go

```diff
@@ -947,6 +947,12 @@ func resolveFK(
 			}
 		}
 		if !found {
+			// Avoid unexpected index builds from ALTER TABLE ADD CONSTRAINT.
+			if mode == sqlbase.ConstraintValidity_Unvalidated {
```

Comment:
I'm confused as to why this error is related to the mode. When does this mode get set, and why is unsetting it okay?